examples/js/modifiers/BufferSubdivisionModifier.js

```diff
@@ -0,0 +1,766 @@
+ *		http://www.holmes3d.net/graphics/subdivision/
+ *		http://www.cs.rutgers.edu/~decarlo/readings/subdiv-sg00c.pdf
+ *
+ *	Known Issues:
+ *		- currently doesn't handle "Sharp Edges"
+ *      - no checks to prevent breaking when uv's don't exist.
+ *      - vertex colors are unsupported.
+ *     **DDS Images when using corrected uv's passed to subdivision modifier will have their uv's flipy'd within the correct uv set
+ *     **Either flipy the DDS image, or use shaders. Don't try correcting the uv's before passing into subdiv (eg: v=1-v).
+ *
+ * @input THREE.Geometry, or index'd THREE.BufferGeometry with faceUV's (Not vertex uv's)
+ * @output non-indexed vertex points, uv's, normals.
+ *     
+ */
+
+/*
+ *
+ * The TypedArrayHelper class is designed to assist managing typed arrays, and to allow the removal of all 'new Vector3, new Face3, new Vector2'.
+ * 
+ * It will automatically resize them if trying to push a new element to an array that isn't long enough
+ * It provides 'registers' that the units can be mapped to. This allows a small set of objects
+ * (ex: vector3's, face3's, vector2's) to be allocated then used, to eliminate any need to rewrite all
+ * the features those classes offer while not requiring some_huge_number to be allocated.
+ * It should be moved into it's own file honestly, then included before the BufferSubdivisionModifier - maybe in three's core?
+ *
+ *
+ * EX: new TypedArrayHelper(initial_size_in_elements, 3, THREE.Vector3, Float32Array, 3, ['x', 'y', 'z']); (the x,y,z comes from THREE.Vector3. It would be abc if it were a face3. etc etc)
+ *
+ */
+THREE.Face3.prototype.set = function (a, b, c) {
+    this.a = a;
+    this.b = b;
+    this.c = c;
+}
+var TypedArrayHelper = function (size, registers, register_type, array_type, unit_size, accessors) {
+    this.array_type = array_type;
+    this.register_type = register_type;
+    this.unit_size = unit_size;
+    this.accessors = accessors;
+    this.buffer = new array_type(size * unit_size);
+    this.register = [];
+    this.length = 0;
+    this.real_length = size;
+    this.available_registers = registers;
+    for (var i = 0; i < registers; i++) { this.register.push(new register_type()); }
+}
+TypedArrayHelper.prototype = {
+    constructor: TypedArrayHelper,
+    index_to_register: function (index, register, isLoop) {
+        var base = index * this.unit_size;
+        if (register >= this.available_registers) {
+            throw ("Nope nope nope, not enough registers!");
+        }
+        if (index > this.length) {
+            throw ("Nope nope nope, index is out of range");
+        }
+        for (var i = 0; i < this.unit_size; i++)
+            (this.register[register])[this.accessors[i]] = this.buffer[base + i];
+
+    },
+    resize: function (new_size) {
+        if (new_size == 0) new_size = 8;
+        var nBuffer = new this.array_type(new_size * this.unit_size);
+        nBuffer.set(this.buffer.slice(0, this.length * this.unit_size));
```

Comment:
[TypedArray.prototype.slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/slice) does not appear to be implemented in all browsers. So you should probably address this issue before merge.

And as said, also needlessly doubles your peak memory consumption. Doesn't matter how often it's run, may still be just enough to hit your browser's limit on a large mesh.